apis/v1beta1/object_reference_types.go

```diff
@@ -46,8 +46,8 @@ type LocalObjectReference struct {
 // be rejected by the implementation, with appropriate Conditions set
 // on the containing object.
 type SecretObjectReference struct {
-	// Group is the group of the referent. For example, "networking.k8s.io".
-	// When unspecified (empty string), core API group is inferred.
+	// Group is the group of the referent. For example, "gateway.networking.k8s.io".
+	// When unspecified or empty string, core API group is inferred.
```

Comment:
This is going to be interesting to see in patch updates. Consider this

create w/ Group: "G"
update other fields w/ Group field value: "G"
Update other fields w/Group field value:"" // which will be defaulted to "<whatever>" removing "G" which is definitly not the intention. Can we double check the behavior?

Reply:
I tried this out with the following transition: 

1.
```yaml
    tls:
      certificateRefs:
      - kind: Secret
        group: "g"
        name: foo-example-com-cert
```

2. 
```yaml
    tls:
      certificateRefs:
      - kind: Secret
        name: foo-example-com-cert
```

As you expected, after applying step 2, the group became "" instead of "g". With that said, I'm not sure why you would remove a group from a manifest, or if you were trying to just update a subset, why you wouldn't use patch. This is likely something we need to more clearly document, but I'm hoping it doesn't represent a typical workflow.